### PR TITLE
New package: ecl-16.1.3

### PR DESCRIPTION
**Pull request by knusbaum**

### CODE REVIEW

File: srcpkgs/ecl/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
+		vmove usr/lib/ecl-${version}/*.a
+	}
+}
+
+post_install() {
```
Comment (maxice8) [inline srcpkgs/ecl/template]: This comes before subpackage definitions

File: srcpkgs/ecl/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'ecl'
+pkgname=ecl
+version=16.1.3
+revision=1
+build_style=gnu-configure
+hostmakedepends="m4"
```
Comment (maxice8) [inline srcpkgs/ecl/template]: hostmakedepends should include only `pkg-config` for this case

File: srcpkgs/ecl/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'ecl'
+pkgname=ecl
+version=16.1.3
+revision=1
+build_style=gnu-configure
```
Comment (maxice8) [inline srcpkgs/ecl/template]: add this line directly below build_style

```
configure_args="--enable-gmp=system --enable-boehm=system --enable-libatomic=system"
```

File: srcpkgs/ecl/template (hunk 2 above)
Comment (maxice8) [inline srcpkgs/ecl/template]: add this line directly below hostmakedepends=

```
depends="gc-devel libatomic_ops-devel gmp-devel"
```

File: srcpkgs/ecl/template [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+distfiles="https://common-lisp.net/project/ecl/static/files/release/ecl-${version}.tgz"
+checksum=76a585c616e8fa83a6b7209325a309da5bc0ca68e0658f396f49955638111254
+
+ecl-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
+		vmove usr/lib/ecl-${version}/*.a
```
Comment (maxice8) [inline srcpkgs/ecl/template]: please quote globbing

File: srcpkgs/ecl/template [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="LGPL-2.1-or-later"
+homepage="https://common-lisp.net/project/ecl/"
+distfiles="https://common-lisp.net/project/ecl/static/files/release/ecl-${version}.tgz"
+checksum=76a585c616e8fa83a6b7209325a309da5bc0ca68e0658f396f49955638111254
+
+ecl-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
```
Comment (maxice8) [inline srcpkgs/ecl/template]: add

```
vmove usr/bin/ecl-config
vmove usr/share/man/man1/ecl-config.1
```

File: srcpkgs/ecl/template [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+short_desc="A Common-Lisp interpreter as described in the X3J13 Ansi specification"
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="LGPL-2.1-or-later"
+homepage="https://common-lisp.net/project/ecl/"
+distfiles="https://common-lisp.net/project/ecl/static/files/release/ecl-${version}.tgz"
+checksum=76a585c616e8fa83a6b7209325a309da5bc0ca68e0658f396f49955638111254
+
+ecl-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
```
Comment (maxice8) [inline srcpkgs/ecl/template]: add $makedepends to depends= here

File: srcpkgs/ecl/template (hunk 2 above)
Comment (knusbaum) [reply srcpkgs/ecl/template]: I think this is `makedepends=...` yes?

Comment (maxice8) [reply srcpkgs/ecl/template]: yes, makedepends=, sorry it was late

**Comment by knusbaum** [comment]
@maxice8 Thanks for figuring that out so quickly. I planned to figure it out last night. How did you figure out the issues so fast? The build was working locally, even in a clean chroot, so I was surprised to see travis fail like that.

**Comment by maxice8** [comment]
> @maxice8 Thanks for figuring that out so quickly. I planned to figure it out last night. How did you figure out the issues so fast? The build was working locally, even in a clean chroot, so I was surprised to see travis fail like that.

problably experience you start developing an eye for looking out what's wrong in a template the more you work with them

**Comment by knusbaum** [comment]
@maxice8 It seems like for cross-compiling, there needs to be a version of ecl installed on the host. If the program is being compiled for the same architecture as the host, it doesn't need it, but for cross-compiling it does.

I'm not sure how best to handle this. I could try building and installing the program on the host post extraction, but that seems messy. It works if I just put ecl in the hostdepends (for cross-compiles), so long as I've run `./xbps-src pkg ecl` before `./xbps-src -a aarch64 pkg ecl`

Do you have a recommendation?

**Comment by maxice8** [comment]
> t doesn't need it, but for cross-compiling it does.

```
if [ "$CROSS_BUILD" ]; then
    hostmakedepends+=" ecl-devel"
fi
```

see `gopass` template

File: srcpkgs/ecl/template (hunk 3 above)
Comment (knusbaum) [reply srcpkgs/ecl/template]: done.

File: srcpkgs/ecl/template [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="LGPL-2.1-or-later"
+homepage="https://common-lisp.net/project/ecl/"
+distfiles="https://common-lisp.net/project/ecl/static/files/release/ecl-${version}.tgz"
+checksum=76a585c616e8fa83a6b7209325a309da5bc0ca68e0658f396f49955638111254
+
+if [ "$CROSS_BUILD" ]; then
+	# Depend on system ecl
+	hostmakedepends+=" ecl"
+	configure_args+=" --with-cross-config=/void-packages/srcpkgs/ecl/files/${XBPS_TARGET_MACHINE}.conf"
```
Comment (maxice8) [inline srcpkgs/ecl/template]: use XBPS_SRCPKGDIR

**Comment by knusbaum** [comment]
I removed the devel package after discovering that significant portions of ecl are broken without it installed. It doesn't make sense to have them as separate packages.

**Comment by knusbaum** [comment]
@maxice8 Is there anything else I need to do for this?

**Pull request merged**